### DISCUSSION TITLE
Welcome to DarkerNodes Discussions!

### DISCUSSION THREAD

**Discussion by hugoattal** [TheHerobrine/DarkerNodes#1] [General]
Let's try this new feature of GitHub, it's meant for your questions/suggestions/ideas for Darker Nodes 🥳!

**Comment by gradgr** [comment]
Hey mate! Great Job!

i have posted on Questions tab on marketplace, though i guess this is more appropriate to discuss here

- Comments on BPs are unreadable ( white font)
- Also many text fonts are in black and cannot be read unless marked with cursor
- Please make upper color line from GET/SET nodes a bit thicker as it is very thin to be seen clearly
- Selecting BP components from the list is a bit hard to read ( black font )
- Selection of assets on library wont respect PrimaryColor Selection (orange)

Thank you!

**Comment by hugoattal** [comment]
Hey @gradgr, thank you for spotting these 😄!

> Comments on BPs are unreadable ( white font)

Yes, I made an issue on that, this will be quickly fixed: https://github.com/TheHerobrine/DarkerNodes/issues/8

> Also many text fonts are in black and cannot be read unless marked with cursor

Yes, I also made an issue, but there might some that are hard coded and that I cannot change thought... https://github.com/TheHerobrine/DarkerNodes/issues/3

> Please make upper color line from GET/SET nodes a bit thicker as it is very thin to be seen clearly

Ok, I'll add a setting to change that! I've created an issue: https://github.com/TheHerobrine/DarkerNodes/issues/14

> Selecting BP components from the list is a bit hard to read (black font)

Maybe you'll need an editor restart to fix that. The primary color needs two restarts to be set (Yeah, it's not that optimal, I didn't find another way yet)

> Selection of assets on library wont respect PrimaryColor Selection (orange)

Same thing, you might need a restart.

Please tell me if that fixes the last two request 🙂!

**Comment by gradgr** [comment]
Yeah mate 2nd restart worked :-) weird though 

{copy pasting question from marketplace}

-Additional Options :

i. change node style?

ii. remove the x,y axis in blueprints that are quite annoying...

Bugs/To DO :

i. many UI icons are missing at the moment

ii. activated(selected) tab overlaps ontop of others , quite annoying :-)

**Comment by hugoattal** [comment]
> change node style?

You mean back to default? Or another node style?

> remove the x,y axis in blueprints that are quite annoying...

It's already available under `Edit/Editor Preferences/Plugins/Darker Nodes/Use Blueprint Grid` 😉

> i. many UI icons are missing at the moment

Yes, I know, I won't do it all, but I'll add more and more!

> ii. activated(selected) tab overlaps ontop of others , quite annoying :-)

It's solved here 😉 https://github.com/TheHerobrine/DarkerNodes/issues/10

**Reply by gradgr** [reply]
No like more different styles from minimal to more detailed.
No this option will remove completely the grid, i mean disable the 0,0 axis lines

**Reply by hugoattal** [reply]
> No like more different styles from minimal to more detailed.

Oh ok, I'll add some settings to change the way they are display.

> No this option will remove completely the grid, i mean disable the 0,0 axis lines

Well, either it removes completely the grid, either it only removes the x=0 and the y=0 axis 😅

**Comment by Frustrated-Programmer** [comment]
I don't own this plugin but def plan on getting it as soon as I get some extra $.

How possible is it for like a preview screen?  I've seen from a review that you have to restart just to view your changes. Is it possible for like a small window on the side that shows what the new theme _would_ look like?

**Reply by hugoattal** [reply]
Hey @Frustrated-Programmer ! Well, I'm trying to make it work without a restart (as a live update), it might actually be possible!
I've created an issue for that: https://github.com/TheHerobrine/DarkerNodes/issues/20

On the other hand, a preview screen seems a bit too complicated, that's really not an easy thing to do with the engine 😕...

**Comment by S-ed** [comment]
@TheHerobrine Nice!
I've tried to make a minimalistic theme 3 years ago. Can be found on my UE4 fork.
But I was not aware you can modify the Slate rendering from the plugin. That makes "Theme engine" possible?
What are the limitations of it? Can nodes be rewritten? (the Slate part) Or it's just changes the Texture sheet ue4 generating from PNGs?

**Reply by hugoattal** [reply]
Thank you 🙂!

Well, it's really not that straight forward, I had to reverse engineering it a bit and hack around a few times...
Not everything is possible, some things are hard-coded (and you really can't do anything about it)

You can change the textures, the fonts and some margin/padding. But the worst part is figuring out where to change that specific texture/color 😅... For some elements, I spent a few hours to find where to modify it (and sometimes figuring out that it was hard-coded and that I can't do anything about it).

That's why I tried to make a really comprehensive setting panel for people to customize it easily as they want:
![image](https://user-images.githubusercontent.com/4563971/102703253-c4c3ce80-426c-11eb-80b3-a44cd121b3cc.png)

I'm also working on a light theme, but it's a nightmare 😅...

So to answer your questions directly:

> That makes "Theme engine" possible?

Well, that's basically Darker Nodes 😛!

> What are the limitations of it?

Clearly you can't do everything you want, it's reaaaally time-consuming and a bug magnet.

> Can nodes be rewritten?

I'm not sure to understand it correctly, but if you meant "change the slate structure of the nodes in blueprint", I really don't think you can. (unless you find a filthy  C++ hack I'm not aware of (and I'm aware of a lot of hacks))

> Or it's just changes the Texture sheet ue4 generating from PNGs?

Well, it's basically just texture change.

---

If you wish to see some features/settings exposed in Darker Nodes, don't hesitate to create an issue for that 🙂!

**Comment by raijoking** [comment]
Did the 1.4 updates is already released?

**Reply by hugoattal** [reply]
@raijoking Yes it is (on both 4.25 and 4.26 unreal version). If you don't see it, maybe restart the Epic Game launcher 🤔...

**Reply by hugoattal** [reply]
You can check your current version number under Edit/Plugins:
![image](https://user-images.githubusercontent.com/4563971/102729260-4af51900-4330-11eb-9a62-0bc935ac748e.png)

**Reply by raijoking** [reply]
Thank you :) I already updated it :)

**Reply by s-t-a-y** [reply]
How do I update the Plugins from the Editor?

**Reply by hugoattal** [reply]
@s-t-a-y I'm sorry, but you must update the plugin from the launcher (closing the editor before), it's how Epic makes it work 😕

**Comment by Saiodin** [comment]
Heyho,

I like to customize the grid colors to make it only slightly visible. Unfortunately Darker Nodes overwrites these values. Can this be added as an option in the plugin instead? Or not touch them at all if the "Use Blueprint Grid" function is checked?

Cheers!
![image](https://user-images.githubusercontent.com/14908776/106822383-56519900-667f-11eb-9d37-c801ef19744a.png)

**Reply by hugoattal** [reply]
Sorry, I totally miss your message! Ok, I understand, I'll add an option to customize them 😉
https://github.com/TheHerobrine/DarkerNodes/issues/130

(I will be tackling all the issues of Darker Nodes this weekend to have an update for Monday)
